app/components/file_attachment_list/file_attachment_image.js

```diff
@@ -168,22 +119,27 @@ export default class FileAttachmentImage extends PureComponent {
             imageStyle = {height, width, position: 'absolute', top: 0, left: 0, borderBottomLeftRadius: 2, borderTopLeftRadius: 2};
         }
 
+        const imageProps = {};
+        if (file.localPath) {
+            imageProps.defaultSource = {uri: file.localPath};
+        } else {
+            imageProps.thumbnailUri = Client4.getFileThumbnailUrl(file.id);
+            imageProps.imageUri = Client4.getFilePreviewUrl(file.id);
+        }
+
         return (
-            <View style={[style.fileImageWrapper, {backgroundColor: wrapperBackgroundColor, height: wrapperHeight, width: wrapperWidth, overflow: 'hidden'}]}>
-                <AnimatedView style={{height: imageHeight, width: imageWidth, backgroundColor: wrapperBackgroundColor, opacity}}>
-                    <Image
-                        style={imageStyle}
-                        source={source}
-                        resizeMode={resizeMode}
-                        resizeMethod={resizeMethod}
-                        {...imageComponentLoaders}
-                    />
-                </AnimatedView>
-                {(!isInFetchCache && !file.failed && (file.loading || this.state.requesting)) &&
-                <View style={[style.loaderContainer, {backgroundColor: loadingBackgroundColor}]}>
-                    <ActivityIndicator size='small'/>
-                </View>
-                }
+            <View
+                ref={this.handleCaptureRef}
```

Comment:
I'm not so sure about passing refs around like this, but it does seem way nicer than adding a `this.getView` method that gets broken by connecting the component to the redux store

Reply:
this works great